APIKit/APIKit.swift

```diff
@@ -26,16 +26,45 @@ public enum Method: String {
     case CONNECT = "CONNECT"
 }
 
-public class API {
-    // configurations
-    public class func baseURL() -> NSURL {
-        return NSURL()
-    }
+private var dataTaskResponseBufferKey = 0
+private var dataTaskCompletionHandlerKey = 0
 
-    public class func URLSession() -> NSURLSession {
-        return NSURLSession.sharedSession()
+private extension NSURLSessionDataTask {
+    private var responseBuffer: NSMutableData {
+        if let responseBuffer = objc_getAssociatedObject(self, &dataTaskResponseBufferKey) as? NSMutableData {
+            return responseBuffer
+        } else {
+            let responseBuffer = NSMutableData()
+            objc_setAssociatedObject(self, &dataTaskResponseBufferKey, responseBuffer, UInt(OBJC_ASSOCIATION_RETAIN_NONATOMIC))
+            return responseBuffer
+        }
+    }
+    
+    private var completionHandler: ((NSData!, NSURLResponse!, NSError!) -> Void)? {
```

Comment:
I think this signature should be `((NSData, NSURLResponse?, NSError?) -> Void)?` to avoid using IUO.